core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -160,6 +162,20 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     set("spark.serializer", classOf[KryoSerializer].getName)
     this
   }
+  /**
+   * Use Kryo serialization and register the given set of Avro schemas so that the generic
+   * record serializer can decrease network IO
+   */
+  def registerAvroSchemas(schemas: Schema*): SparkConf = schemas.foldLeft(this) { (conf, schema) =>
+    conf.set(avroSchemaKey(schema), schema.toString)
+  }
+
+  /** Gets all the avro schemas in the configuration used in the generic Avro record serializer */
+  def getAvroSchema: Map[Long, String] =
+    getAll.filter { case (k, v) => k.startsWith(avroSchemaNamespace) }
+          .map { case (k, v) => (k.substring(avroSchemaNamespace.length).toLong, v) }
+          .toMap
+
```

Comment:
nit: I think this method should be wrapped in braces since its multiline, (even though its one statement)

also, delete extra newline